Add english docs for UIE (#4255)
PaddlePaddle/PaddleNLP: applications/information_extraction/text/data_distill/README_en.md

# UIE Slim data distillation

> Behind the powerful extraction capability of UIE, it also requires a large computing power to support calculations. In some industrial application scenarios, the performance requirements are high, and it cannot be practically applied without effective compression. Therefore, we built the UIE Slim data distillation system based on the data distillation technology. The principle is to use the data as a bridge to transfer the knowledge of the UIE model to the small closed-domain information extraction model, so as to achieve the effect of greatly improving the prediction speed with a small loss of accuracy.

Reviewer: ```suggestion
While UIE has powerful zero-shot extraction capabilities, its prompting structure requires significant compute to serve in real time. Some industrial application scenarios have high inference performance requirements and the model cannot go into production without being effectively compressed. We built the UIE Slim Data Distillation with knowledge distillation techniques. The principle is to use the data as a bridge to transfer the knowledge of the UIE model to the smaller closed-domain information extraction model in order to achieve speedup inference significantly with minimal loss to accuracy.
```

#### Three steps of UIE data distillation

> - **Step 1**: Use the UIE model to finetune the labeled data to get the Teacher Model.
> 
> - **Step 2**: The user provides large-scale unlabeled data, which needs to be from the same source as the labeled data. Prediction on unsupervised data using Taskflow UIE.
> 
> - **Step 3**: Use the labeled data and the synthetic data obtained in step 2 to train a closed-domain Student Model.

Reviewer: ```suggestion
- **Step 1**: Finetune the UIE model on the labeled data to get the Teacher Model.

- **Step 2**: Process the user-provided unlabeled data and run inference with Taskflow UIE.

- **Step 3**: Use the labeled data and the inference results obtained in step 2 to train a closed-domain Student Model.
```

     --save_dir student_data \
     --task_type relation_extraction \
     --synthetic_ratio 10 \
     --model_path ../checkpoint/model_best
```

**NOTE**: The schema needs to be configured in `data_distill.py` according to the label data, and the schema needs to contain all label types in the label data.

Description of configurable parameters:

- `data_path`: Path to labeled data (`doccano_ext.json`) and unsupervised text (`unlabeled_data.txt`).
- `model_path`: The path of the trained UIE custom model.
- `save_dir`: The path to save the training data of the student model.
- `synthetic_ratio`: Controls the ratio of synthetic data. The maximum number of synthetic data=synthetic_ratio*number of labeled data.
- `platform`: The labeling platform used to label data, optional are `doccano`, `label_studio`, the default is `label_studio`.
- `task_type`: Select the task type, optional are `entity_extraction`, `relation_extraction`, `event_extraction` and `opinion_extraction`. Because it is a closed-domain extraction, the post-processing logic of different tasks is different, so the task type needs to be specified.
- `seed`: random seed, default is 1000.

#### Teacher model evaluation

> In the UIE fine-tuning stage, the model effect is evaluated on UIE training format data (this evaluation method is not an end-to-end evaluation method, not a standard evaluation method for relation extraction or event extraction), and the end-to-end evaluation can be performed through the following evaluation script.

Reviewer: ```suggestion
In the UIE fine-tuning stage, the model performance is evaluated on UIE training format data, which is not a standard end-to-end evaluation method for relation extraction or event extraction. The end-to-end evaluation can be performed through the following evaluation script.
```
